STX.REST.RESTFulSense.Clients.Tests.Unit/Services/Foundations/HttpExchanges/HttpExchangeServiceTests.Headers.cs

```diff
@@ -0,0 +1,1787 @@
+        {
+            return Enumerable.Range(0, GetRandomNumber())
+                .Select(item => CreateRandomViaHeader())
+                .ToArray();
+        }
+
+        private static dynamic CreateRandomViaHeader()
+        {
+            return new
+            {
+                ProtocolName = GetRandomString(),
+                ProtocolVersion = GetRandomString(),
+                ReceivedBy = GetRandomString(),
+                Comment = $"({GetRandomString()})",
+            };
+        }
+
+        private static dynamic[] CreateRandomWarningHeaderArray()
+        {
+            return Enumerable.Range(0, GetRandomNumber())
+                .Select(item => CreateRandomWarningHeader())
+                .ToArray();
+        }
+
+        private static dynamic CreateRandomWarningHeader()
+        {
+            return new
+            {
+                Code = GetRandomNumber(),
+                Agent = GetRandomString(),
+                Text = CreateRandomQuotedString(),
+                Date = GetRandomDateTime(),
+            };
+        }
+
+        private static dynamic CreateRandomRetryConditionHeader()
+        {
+            bool isEven = GetRandomNumber() % 2 == 0;
+
+            return new
+            {
+                Date = isEven ? GetRandomDateTime() : default(DateTimeOffset?),
+                Delta = isEven ? default(TimeSpan?) : GetRandomTimeSpan()
+            };
+        }
+
+        private static dynamic[] CreateRandomProductHeaderArray()
+        {
+            return Enumerable.Range(0, GetRandomNumber())
+                .Select(item => CreateRandomProductHeader())
+                .ToArray();
+        }
+
+        private static dynamic CreateRandomContentDispositionHeader()
+        {
+            return new
+            {
+                DispositionType = GetRandomString(),
+                Name = GetRandomString(),
+                FileName = GetRandomString(),
+                FileNameStar = GetRandomString(),
+                CreationDate = GetRandomDateTimeWithOutFractions(),
+                ModificationDate = GetRandomDateTimeWithOutFractions(),
+                ReadDate = GetRandomDateTimeWithOutFractions(),
+                Size = GetRandomLong(),
+            };
+        }
+
+        private static dynamic CreateRandomRangeContentHeader()
+        {
+            long from = GetRandomLong();
+            long to = CreateRandomLongToValue(from);
+            long length = to;
+
+            return new
+            {
+                Unit = GetRandomString(),
+                From = from,
+                To = to,
+                Length = length
+            };
+        }
+
+        private static dynamic CreateAcceptHeaderException(MediaTypeHeader[] invalidAcceptHeader)
+        {
+            var invalidHttpExchangeHeaderException = new InvalidHttpExchangeHeaderException(
+              message: "Invalid HttpExchange request header error occurred, fix errors and try again.");
```

Comment:
@glhays, Presently, the model is used for both the request header and the content header. so the message string can be either `"Invalid HttpExchange request header error occurred, fix errors and try again."` or `"Invalid HttpExchange content header error occurred, fix errors and try again."`. that was the reason for the generic method signature.
do you think we should create two separate models for the content header errors and request header errors?

what do you think? @glhays @LBoullosa

Reply:
maybe we could leave "Invalid HttpExchange header....." and then as the details specifies if it is a content or request could be OK

Reply:
Sounds good.

Reply:
No it can't be either or. Is that Standard compliant, if so then in that case just create one generic exception and just create as many string messages that fit the specific scenario. Come on guys what's going on here?

Reply:
> No it can't be either or. Is that Standard compliant, if so then in that case just create one generic exception and just create as many string messages that fit the specific scenario. Come on guys what's going on here?

this is exactly what I was arguing from begining. A generic exception and many messages depending on the case.